bindings-go/apis/v2/cdutils/resources.go

```diff
@@ -42,3 +43,20 @@ func GetImageReferenceByName(cd *cdv2.ComponentDescriptor, name string) (string,
 	}
 	return ociImageAccess.ImageReference, nil
 }
+
+// GetRepositoryAndTagFromReference takes an image reference
+// e.g eu.gcr.io/gardener-project/gardener/gardenlet:v1.11.3
+// and returns the image repository as a first, and the tag as a second argument
+func GetRepositoryAndTagFromReference(imageReference string) (string, string, error) {
```

Comment:
you should also consider hashes liek `eu.gc.rio/gardener-project/gardenlet@sha256:12345`